### DISCUSSION TITLE
Read between special comments as well as a visual block

### DISCUSSION THREAD

**Discussion by ryuheechul** [michaelb/sniprun#148] [Ideas]
with an example of below
```js
// other code above...

// code that I want to run with SnipRun is below

const a = 1
console.log(a + b)

// code that I want to run with SnipRun is above

// other code below...
```

I could select the visual block of just
```js
const a = 1
console.log(a + b)
```

Which is fine when I only run it for few times and the lines that I'm selecting is just few lines.

However if I would like to repeat this process, selecting the same (potential many) lines again and again would be undesirable.

So I propose an idea of using special comment for SnipRun to figure out the lines to run

```js
// other code above...

// sniprun:[
const a = 1
console.log(a + b)
// sniprun:]

// other code below...
```

_the naming of the special comment is arbitrary choice from just on top of my head (so please suggest any better ones!)_

So whenever the cursor is in between `// sniprun:[` and `// sniprun:]`, `:SnipRun` just figure out which portion of the code to run without having to provide visual selection everytime
(additionally the plugin can highlight the special comment so users notice what they means as well)

**Comment by michaelb** [comment]
Nice idea!

There's already something similar for literate programming (markdown and org-mode), so this shouldn't be too hard to make

However, I'm a bit conflicted about using a comment, I mean, I don't want people to keep and commit // sniprun in their VCS....
It'd feel like an ad, and would be a big code smell


I think we should discuss the technical design a bit more (as well as the behavior you'd like)

How do marks (can they be made persitent?) and virtual text indicators  sound ?

**Reply by ryuheechul** [reply]
I personally wouldn’t commit the comments as it should be a temporary running but however it’s possible anyone would do that.

I actually haven’t used marks in vim yet so can’t comment on that just now, but I will give it a try and let you know!
